### PR TITLE
Implement UCS2 checks

### PR DESCRIPTION
**Pull request by rmcgibbo**

### CODE REVIEW

**Comment by rmcgibbo** [comment]
@njsmith, I'm just going to add `--no-check-certificate`, but this may or may not be a misconfiguration of your server:

```
Connecting to vorpus.org (vorpus.org)|69.93.127.174|:443... connected.
ERROR: cannot verify vorpus.org's certificate, issued by `/C=XX/ST=There is no such thing outside US/L=Everywhere/O=OCOSA/OU=Office for Complication of Otherwise Simple Affairs/CN=roberts.vorpus.org/emailAddress=root@roberts.vorpus.org':
  Self-signed certificate encountered.
    ERROR: certificate common name `roberts.vorpus.org' doesn't match requested host name `vorpus.org'.
To connect to vorpus.org insecurely, use `--no-check-certificate'.
```

**Comment by njsmith** [comment]
Yeah, misconfiguration, sorry -- use `https://vorpus.org` instead of `https://roberts.vorpus.org`

**Comment by rmcgibbo** [comment]
The code is using `https://vorpus.org` https://github.com/manylinux/auditwheel/blob/usc2/tests/test-wheelhouses.sh#L11-L12

File: auditwheel/policy/external_references.py [hunk 1]
```diff
@@ -13,8 +13,8 @@
 LIBPYTHON_RE = re.compile('^libpython\d\.\dm?.so(.\d)*$')
 
 
-def elf_exteral_referenences(elftree: Dict, wheel_path: str):
-    # XXX: Document the elftree structure, or put it in something
+def lddtree_exteral_referenences(lddtree: Dict, wheel_path: str):
```
Comment (njsmith) [inline auditwheel/policy/external_references.py]: FYI `external` and `references` both appear to be misspelled :-)

**Comment by njsmith** [comment]
well, wtf, then, that's weird -- are you somehow using a http client that doesn't do SNI? because you seem to be getting the self-signed certificate that debian autogenerates for unrecognized vhosts, and if you visit that URL in a web browser then you definitely get a real, not-self-signed certificate valid for vorpus.org (and with no mention of roberts.vorpus.org).

...Though, uh. Stepping back, I'm not sure I'm comfortable with my little VPS there becoming critical infrastructure :-)

For tests, maybe it would better to put together a small collection of small modules that break the rules in known, different ways? E.g. a little Cython module built (a) correctly, (b) linking to an illegal library, (c) with bad symbol versions, (d) with a UCS-2 build, and then just check the resulting wheels into the tree?

**Comment by rmcgibbo** [comment]
> ...Though, uh. Stepping back, I'm not sure I'm comfortable with my little VPS there becoming critical infrastructure :-)

Yes, that's fair. The tests I have were kind of just throw together randomly. I did add a wheel satisfying property (d) to the tree in this PR though.

**Comment by rmcgibbo** [comment]
But there are lots of possible corner cases, so there is some virtue in the shotgun approach for testing.

**Comment by rmcgibbo** [comment]
Closes #8
